### PR TITLE
Add restrictions for Prow staging process

### PR DESCRIPTION
**Pull request by chizhg**
**What this PR does, why we need it**:
1. Do not allow developers to directly change production Prow core and cluster configs, as prow-config-updater tool will automatically roll out staging changes to production.
2. prow-auto-bumper tool only updates staging Prow config files.

Plus some fixes to the prow-auto-updater tool:
1. Running the prow-updater-tool requires `bazel`, `gcloud` and `kubectl`, I'm now building the image based on `kubekins-e2e`, will make optimizations by only installing components we need. I have added a TODO.
2. Add the missed `Makefile` in the prow-auto-updater tool
3. In running command to update `testgrid`, use the env vars as it needs the `GOOGLE_APPLICATION_CREDENTIALS` to be set.

### CODE REVIEW

File: tools/prow-config-updater/config/config.go [hunk 1]
```diff
@@ -54,39 +55,42 @@ const (
 
 var (
 	// Prow config root paths.
-	prodProwConfigRoot    = filepath.Join(configPath, string(ProdProwEnv))
-	stagingProwConfigRoot = filepath.Join(configPath, string(StagingProwEnv))
+	ProdProwConfigRoot    = filepath.Join(configPath, string(ProdProwEnv))
+	StagingProwConfigRoot = filepath.Join(configPath, string(StagingProwEnv))
+	// Prow config templates paths.
+	ProdProwConfigTemplatesPath = filepath.Join(configTemplatePath, string(ProdProwEnv))
```
Comment (mattmoor) [inline tools/prow-config-updater/config/config.go]: Format Go code:
```suggestion
	ProdProwConfigTemplatesPath    = filepath.Join(configTemplatePath, string(ProdProwEnv))
```

File: test/presubmit-config-check.sh [hunk 2]
```diff
@@ -18,6 +18,10 @@ source $(dirname $0)/../scripts/library.sh
 
 set -e
 
+trap 'echo "--- FAIL: Directly changing production Prow config files is not allowed, please revert the change."' ERR
```
Comment (chaodaiG) [inline test/presubmit-config-check.sh]: let's use add_trap instead

File: test/presubmit-config-check.sh [hunk 3]
```diff
@@ -18,6 +18,10 @@ source $(dirname $0)/../scripts/library.sh
...
+header "Checking Prow config files are not modified by mistake"
```
Comment (chaodaiG) [inline test/presubmit-config-check.sh]:
```suggestion
header "Checking to make sure Prow productions config files are not modified manually"
```

File: test/presubmit-config-check.sh (hunk 2 above)
Comment (chaodaiG) [reply test/presubmit-config-check.sh]: Also, the error message should be more actionable, like move the changes to staging dir

**Comment by chaodaiG** [comment]
/lgtm

/hold
in case you still have last minute change

File: test/presubmit-config-check.sh (hunk 2 above)
Comment (chizhg) [reply test/presubmit-config-check.sh]: We don't have `add_trap` function in `library.sh`. Do you want me to add it there?

And I changed the error message.

Comment (chaodaiG) [reply test/presubmit-config-check.sh]: I think it's fine for this script, since these 2 traps don't collide with each other

**Pull request merged**